pkg/controller/cloud/node_controller.go

```diff
@@ -343,11 +343,11 @@ func excludeCloudTaint(taints []v1.Taint) []v1.Taint {
 
 // ensureNodeExistsByProviderID checks if the instance exists by the provider id,
 // If provider id in spec is empty it calls instanceId with node name to get provider id
-func ensureNodeExistsByProviderID(instances cloudprovider.Instances, node *v1.Node) (bool, error) {
+func ensureNodeExistsByProviderID(instances cloudprovider.Instances, node *v1.Node, cloud cloudprovider.Interface) (bool, error) {
 	providerID := node.Spec.ProviderID
 	if providerID == "" {
 		var err error
-		providerID, err = instances.InstanceID(context.TODO(), types.NodeName(node.Name))
+		providerID, err = cloudprovider.GetInstanceProviderID(context.TODO(), cloud, types.NodeName(node.Name))
```

Comment:
Thank you for commenting.

> This is originally calling InstanceID for legacy reasons. It only applies if a provider ID isn't already set which shouldn't be the case.

I understand. But I think if the situation that InstanceID can get an instance id, cloud.ProviderName can get a provider name.
because of cloud.ProviderName is a very simple method what return a provider name string only.
So if the situation that a provider id can use `<provider name>://<instance id>`, I think a provider id should be a unified format.

Reply:
@andrewsykim Ah...I see. Sorry, I mistake.
I fixed my code :sweat_smile: